lib/dradis/plugins/pdf_export/exporter.rb

```diff
@@ -1,165 +1,13 @@
-            # transparent(0.5) { stroke_bounds }  # This will stroke on one page
-            text "<b>Author</b>: #{@author}", inline_format: :true
-            text "<b>Email</b>: #{@email}", inline_format: :true
-            text "<b>Date</b>: #{Time.now.strftime('%Y-%m-%d')}", inline_format: :true
-            # transparent(0.5) { stroke_bounds }  # And this will stroke on the next
-          end
-          start_new_page
-        end
-
-        def project_notes
-          draw_header
-
-          text 'Project notes'
-          move_down 20
-
-          @notes.each do |note|
-            fields = note.fields
-            text "<b>#{fields['Title']}</b>", inline_format: true
-            text fields['Description']
-          end
-
-          start_new_page
-        end
-
-        def summary_of_findings
-          draw_header
-
-          text 'SUMMARY OF FINDINGS'
-          move_down 20
-
-          @sorted.each do |note|
-            fields = note.fields
-            text "• #{fields['Title']} (#{fields['CVSSv2']})"
-          end
-
-          start_new_page
-        end
-
-        def detailed_findings
-          draw_header
-
-          text 'DETAILED FINDINGS'
-          move_down 20
-
-          @sorted.each do |note|
-            fields = note.fields
-            text "<b>#{fields['Title']}</b> (#{fields['CVSSv2']})", inline_format: true
-            text fields['Description']
-
-            move_down 20
-            text "<b>Mitigation:</b>", inline_format: true
-            text fields['Mitigation']
-            start_new_page
-          end
-        end
-
-        def tool_list
-          draw_header
-
-          text 'TOOLS USED'
-          move_down 20
-
-          data = [
-            ['Name', 'Description']
-          ]
-
-          data << ['Dradis Framework', "Collaboration and reporting framework\nhttp://dradisframework.org" ]
-
-          table data, header: true, position: :center
-        end
-
-        def outline
-          outline.define do
-            section('Report Content', destination: 2) do
-              page title: 'Summary of Findings', destination: 2
-              page title: 'Tool List', destination: 3
-            end
-          end
-        end
-
-        def draw_header
-          fill_color 'efefef'
-          fill_rectangle [bounds.left-50, bounds.top + 100], bounds.width + 100, 87
-          fill_color '00000'
-
-          box = bounding_box [bounds.left-50, bounds.top+50], :width  => (bounds.width + 100) do
-
-            font "Helvetica"
-            text "Security Assessment Report", align: :center
-            move_down 20
-
-            stroke_color 'dadada'
-            stroke_horizontal_rule
-            stroke_color '000000'
-
-          end
-          move_down 40
-        end
-
-      end
-
       class Exporter < Dradis::Plugins::Export::Base
         def export()
+          if File.file?(options[:template])
+            load options[:template]
```

Comment:
we should be wary of using `load` as it can be dangerous to run a user-created ruby file. we should proabably note that in the documents.

Reply:
Yes, loading a user created ruby file is what the exact purpose here.
Using `require` would load a user created ruby file too.
We are using load to see the difference when the file changes.